sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationSettingJsonSerializer.java

```diff
@@ -0,0 +1,158 @@
+
+        gen.writeStringField(KEY, value.getKey());
+
+        if (!CoreUtils.isNullOrEmpty(value.getLabel())) {
+            gen.writeStringField(LABEL, value.getLabel());
+        }
+
+        gen.writeStringField(CONTENT_TYPE, value.getContentType());
+        gen.writeStringField(ETAG, value.getETag());
+        gen.writeBooleanField(LOCKED, value.isReadOnly());
+
+        gen.writeObjectFieldStart(TAGS);
+        writeTagsProperties(value.getTags(), gen);
+        gen.writeEndObject();
+
+        if (value.getLastModified() != null) {
+            gen.writeStringField(LAST_MODIFIED, value.getLastModified().format(DateTimeFormatter.ISO_DATE_TIME));
+        }
+        gen.writeEndObject();
+    }
+
+    private static void writeSecretReferenceConfigurationSetting(SecretReferenceConfigurationSetting setting,
+        JsonGenerator gen) throws IOException {
+        gen.writeStartObject();
+        gen.writeStringField(URI, setting.getSecretId());
+        gen.writeEndObject();
+        gen.close();
+    }
+
+    private static void writeFeatureFlagConfigurationSetting(FeatureFlagConfigurationSetting setting,
+        JsonGenerator gen) throws IOException {
+        gen.writeStartObject();
+
+        gen.writeStringField(ID, setting.getFeatureId());
+        gen.writeStringField(DESCRIPTION, setting.getDescription());
+        gen.writeStringField(DISPLAY_NAME, setting.getDisplayName());
+        gen.writeBooleanField(ENABLED, setting.isEnabled());
+
+        gen.writeObjectFieldStart(CONDITIONS);
+        gen.writeArrayFieldStart(CLIENT_FILTERS);
+        for (FeatureFlagFilter filter : setting.getClientFilters()) {
+            gen.writeStartObject();
+            gen.writeStringField(NAME, filter.getName());
+            gen.writeObjectFieldStart(PARAMETERS);
+            writeAdditionalProperties(filter.getParameters(), gen);
+            gen.writeEndObject(); // parameters object
+            gen.writeEndObject(); // each filter object
+        }
+        gen.writeEndArray();
+        gen.writeEndObject();
+
+        gen.writeEndObject();
+        gen.close();
+    }
+
+    private static void writeAdditionalProperties(Map<String, Object> properties, JsonGenerator gen)
+        throws IOException {
+        if (CoreUtils.isNullOrEmpty(properties)) {
+            return;
+        }
+
+        for (Map.Entry<String, Object> property : properties.entrySet()) {
+            gen.writeFieldName(property.getKey());
+            gen.writeObject(property.getValue().toString());
+        }
+    }
+
+    private static void writeTagsProperties(Map<String, String> properties, JsonGenerator gen)
+        throws IOException {
+        if (CoreUtils.isNullOrEmpty(properties)) {
+            return;
+        }
+
+        for (Map.Entry<String, String> property : properties.entrySet()) {
+            gen.writeFieldName(property.getKey());
+            gen.writeObject(property.getValue());
+        }
+    }
```

Comment:
Could these be merged into a single method `writeMap`?

Reply:
yes, it could be